auth/src/express/middleware/route-service.ts

```diff
@@ -78,7 +78,17 @@ export default class RouteService extends RouteHandler {
 
     public hash(request: express.Request, response: express.Response): void {
         const toHash = request.body['toHash'];
-        this.sendResponse(true, this.authHandler.toHash(toHash), response);
+        const salt = request.body['salt'];
+        this.sendResponse(true, 'Successful', response, 200, { hash: this.authHandler.toHash(toHash, salt) });
+    }
+
+    public isEquals(request: express.Request, response: express.Response): void {
+        const toHash = request.body['toHash'];
```

Comment:
If I read this code I wonder where you ensure, that `toHash`, ... are really given. This might be null/undefined which is not good for `isEquals`.